sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -247,7 +246,7 @@ class SdkSynchronizer internal constructor(
     override val latestBirthdayHeight: Int get() = processor.birthdayHeight
 
     override suspend fun prepare(): Synchronizer = apply {
-        storage.prepare()
+        // Do nothing; this could likely be removed
```

Comment:
the idea behind having a prepare() method was to make init() lightweight.  How would a suspending initializer impact SDK clients in general?

Reply:
Our SDK initialization is already kind of heavyweight.  The `Initializer` object needs to load the Rust native libraries (which performs disk IO), so I converted `Initializer.new()` to a suspending function a while back.  An `Initializer` is required to construct the `Synchronizer`.

Since clients already need to make a suspending API call (for the `Initializer`), making `SdkSynchronizer` also use a suspending API call seems like a reasonable tradeoff which simplifies both the public API of the SDK and the implementation.  It makes it much easier to reason about thread-safety, since we have so much less mutable state.

Thoughts?